library_generation/owlbot/bin/entrypoint.sh

```diff
@@ -70,7 +70,12 @@ echo "Fixing missing license headers..."
 python3 "${scripts_root}/owlbot/src/fix-license-headers.py"
 echo "...done"
 
-# ensure formatting on all .java files in the repository
+# Ensure formatting on all .java files in the repository.
+# Here we manually set the user.home system variable. Unfortunately, Maven
+# user.home inference involves the /etc/passwd file (confirmed empirically),
+# instead of the presumable $HOME env var, which may not work properly
+# when `docker run`ning with the -u flag because we may incur in users
+# not registered in the container's /etc/passwd file
 echo "Reformatting source..."
-mvn fmt:format -V --batch-mode --no-transfer-progress
+mvn fmt:format -Duser.home="${HOME}" -V --batch-mode --no-transfer-progress
```

Comment:
So the permission issue (the generated files can't be removed by `shutil.rmtree`) is caused by `mvn`?

Reply:
Docker does indeed take ownership of the volumes (i.e. host machine folders) you pass, which makes the `-u` flag necessary. This is a workaround of a new problem that occurs with maven when running the container with an arbitrary user.

The `user.home` workaround is necessary because if you run the docker container with a user not found in the container's `/etc/passwd` file (which is the case when you run `-u $(id -u):$(id -g)`), then the value of `user.home` will be `'?'`, so here we explicitly set it.